File: newsrecomendation/DbModel/src/Article.py
import mongoengine as me


class Article(me.Document):
    author = me.StringField(required=False)
    title = me.StringField(required=True)
    description = me.StringField(required=True)
    url = me.StringField(required=True)
    urlToImage = me.StringField(required=True)
    content = me.StringField(required=True)
    category = me.StringField(required=False)

    meta = {
        'collection': 'articles'
    }

File: web/src/app/article/article.ts
export class Article {
  author: string = ''
  title: string = ''
  description: string = ''
  url: string = ''
  urlToImage: string = ''

  constructor(fields?: Partial<Article>) {
    if (fields) {
      Object.assign(this, fields)
    }
  }
}
